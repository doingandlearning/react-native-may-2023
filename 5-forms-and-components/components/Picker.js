import { Picker, Text } from "@react-native-picker/picker";

export default function CustomPicker({
  options,
  value,
  setValue,
  width = 300,
}) {
  return (
    <Picker selectedValue={value} onValueChange={setValue} style={{ width }}>
      {options.map((option, index) => {
        return (
          <Picker.Item
            label={`${index + 1} - ${option}`}
            value={option}
            key={option}
          />
        );
      })}
    </Picker>
  );
}
