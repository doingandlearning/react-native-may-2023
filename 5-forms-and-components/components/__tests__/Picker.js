import Picker from "../Picker";
import { render } from "@testing-library/react-native";

describe("Picker component renders properly", () => {
  it("The Picker component renders properly", () => {
    expect(
      render(
        <Picker
          options={["Do something", "Don't do something"]}
          value=""
          setValue={() => {}}
        />
      )
    ).toMatchSnapshot();
  });
});
