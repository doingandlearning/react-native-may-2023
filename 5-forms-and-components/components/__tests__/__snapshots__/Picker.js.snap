// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker component renders properly The Picker component renders properly 1`] = `
<View
  style={
    {
      "width": 300,
    }
  }
>
  <RNCPicker
    items={
      [
        {
          "label": "1 - Do something",
          "testID": undefined,
          "textColor": undefined,
          "value": "Do something",
        },
        {
          "label": "2 - Don't do something",
          "testID": undefined,
          "textColor": undefined,
          "value": "Don't do something",
        },
      ]
    }
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={0}
    style={
      [
        {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
